pkg/cli/install_test.go

```diff
@@ -0,0 +1,165 @@
+		},
+		{
+			name: "acorn install --http-endpoint-pattern: Valid",
+			fields: fields{
+				All:    false,
+				Quiet:  false,
+				Output: "",
+			},
+			commandContext: client.CommandContext{
+				ClientFactory: &testdata.MockClientFactory{},
+				StdOut:        w,
+				StdErr:        w,
+				StdIn:         strings.NewReader(""),
+			},
+			args: args{
+				args:   []string{"--http-endpoint-pattern", "{{.App}}-{{.Container}}.{{.ClusterDomain}}"},
+				client: &testdata.MockClient{},
+			},
+			wantErr: false,
+			wantOut: []string{"✔  Installation done"},
+		},
+		{
+			name: "acorn install --http-endpoint-pattern: Invalid",
+			fields: fields{
+				All:    false,
+				Quiet:  false,
+				Output: "",
+			},
+			commandContext: client.CommandContext{
+				ClientFactory: &testdata.MockClientFactory{},
+				StdOut:        w,
+				StdErr:        w,
+				StdIn:         strings.NewReader(""),
+			},
+			args: args{
+				args:   []string{"--http-endpoint-pattern", "{{.Invalid}}-{{.HttpEndpoint}}.{{.Pattern}}"},
+				client: &testdata.MockClient{},
+			},
+			wantErr: true,
+			wantOut: []string{"endpoint pattern is invalid"},
+		},
+		{
+			name: "acorn install with --cluster-domains: Valid",
+			fields: fields{
+				All:    false,
+				Quiet:  false,
+				Output: "",
+			},
+			commandContext: client.CommandContext{
+				ClientFactory: &testdata.MockClientFactory{},
+				StdOut:        w,
+				StdErr:        w,
+				StdIn:         strings.NewReader(""),
+			},
+			args: args{
+				args:   []string{"--cluster-domain", "foo.bar.com"},
+				client: &testdata.MockClient{},
+			},
+			wantErr: false,
+			wantOut: []string{"✔  Installation done"},
+		},
+		{
+			name: "acorn install --lets-encrypt enabled: Valid",
+			fields: fields{
+				All:    false,
+				Quiet:  false,
+				Output: "",
+			},
+			commandContext: client.CommandContext{
+				ClientFactory: &testdata.MockClientFactory{},
+				StdOut:        w,
+				StdErr:        w,
+				StdIn:         strings.NewReader(""),
+			},
+			args: args{
+				args:   []string{"--lets-encrypt", "enabled", "--lets-encrypt-tos-agree", "--lets-encrypt-email=foo@bar.com"},
+				client: &testdata.MockClient{},
+			},
+			wantErr: false,
+			wantOut: []string{
+				"•  You've enabled automatic TLS certificate provisioning with Let's Encrypt",
+				"✔  Installation done"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(*testing.T) {
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+			tt.args.cmd = NewInstall(tt.commandContext)
+			tt.args.cmd.SetArgs(tt.args.args)
+			err := tt.args.cmd.Execute()
+			var out []byte
+			w.Close()
+			if err != nil {
+				if !tt.wantErr {
+					assert.Failf(t, "got err when err not expected", "got err: %s", err.Error())
+				}
+				out = []byte(err.Error())
+			} else {
+				out, _ = io.ReadAll(r)
+			}
+
+			for _, wantOut := range tt.wantOut {
+				testOut, err := os.ReadFile(wantOut)
+				if err == os.ErrNotExist {
+					testOut = []byte(wantOut)
+				}
```

Comment:
It seems that we are never expecting the `wantOut` here to be a file. I think it would be good if this was simplified.

Reply:
Yeah this and all the other comments you hard a definitely valid. The main reason I structured them this was is in the case that we need to edit these values in the future. However, if we do end up needing to edit the values we can update these tests at that time.